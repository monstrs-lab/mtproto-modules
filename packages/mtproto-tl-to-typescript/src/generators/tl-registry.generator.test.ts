import { tmpdir }               from 'node:os'
import { join }                 from 'node:path'

import { describe }             from '@jest/globals'
import { expect }               from '@jest/globals'
import { it }                   from '@jest/globals'
import { Project }              from 'ts-morph'
import { ScriptTarget }         from 'ts-morph'
import { ModuleKind }           from 'ts-morph'
import { ModuleResolutionKind } from 'ts-morph'

import { TLRegistryGenerator }  from './tl-registry.generator.js'

describe('mtproto tl to typescript', () => {
  describe('tl registry generator', () => {
    it('check generate registry file', async () => {
      const project = new Project({
        compilerOptions: {
          outDir: tmpdir(),
          module: ModuleKind.ES2022,
          target: ScriptTarget.ES2022,
          moduleResolution: ModuleResolutionKind.Bundler,
        },
      })

      project.createSourceFile(join(tmpdir(), 'test.ts')).addClass({
        isExported: true,
        name: 'Test',
        properties: [
          {
            name: 'constructorId',
            type: 'number',
            initializer: '0x00000000',
          },
        ],
      })

      new TLRegistryGenerator(project).generate()

      expect(project.getSourceFile(join(tmpdir(), 'registry.ts'))?.getFullText()).toMatchSnapshot()
    })
  })
})
