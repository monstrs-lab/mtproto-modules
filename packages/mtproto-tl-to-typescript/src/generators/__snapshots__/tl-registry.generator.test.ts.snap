// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mtproto tl to typescript tl registry generator check generate registry file 1`] = `
"import type { TLObject } from "@monstrs/mtproto-tl-core";
import { TLRegistry } from "@monstrs/mtproto-tl-core";

export const registry = new TLRegistry(new Map<number, typeof TLObject>([[0x00000000, async () => (await import('./test.js')).Test]]));
"
`;
